### PR TITLE
Fix memleak in signal package

### LINKED ISSUE #1067: Memory Leak in statusd
# Problem

There appear to be a memory leak in `statusd` that manifests itself mostly in the Digital Ocean fleet since it's more utilized.
Here's an example of hosts running out of memory:
memory_leak
https://grafana.status.im/d/YFtG6HViz/hosts-dashboard?orgId=1&from=1530101258724&to=1530173383509&var-host=node-.*&var-fleet=eth.beta&var-fleet=eth.test&var-dc=do-ams3
It happens roughly every

# Acceptance Criteria

Memory leak is identified and fixed.

# Notes

In relation to this additional metrics were exposed via geth_exporter in:
https://github.com/status-im/status-go/issues/1042

### PR DESCRIPTION
**Pull request by divan**
We were investigating memory leak in a server instance of statusd, and here is a brief explanation of findings: 
https://github.com/status-im/status-go/issues/1067#issuecomment-407567649

TLDR: 
 a) we call signals even when running as a server
 b) we historically never free memory allocated by `C.CString`.

I believe this fix should stop memory from growing (or at least notable decrease growth).

Closes #1067

### CODE REVIEW

File: signal/signals.go [hunk 1]
```diff
@@ -40,8 +41,12 @@ func send(typ string, event interface{}) {
 	data, err := json.Marshal(&signal)
 	if err != nil {
 		logger.Error("Marshalling signal envelope", "error", err)
+		return
 	}
-	C.StatusServiceSignalEvent(C.CString(string(data)))
+
+	str := C.CString(string(data))
+	C.StatusServiceSignalEvent(str)
+	C.free(unsafe.Pointer(str))
```
Comment (adambabik) [inline signal/signals.go:49]: Have you verified that we don't cause any invalid memory reads on Android and iOS? For example, in Android, `C.StatusServiceSignalEvent` we do `if (javaJsonEvent != NULL) (*env)->DeleteLocalRef(env, javaJsonEvent);`.

**Review by adambabik — APPROVED** [review]
Great job!

File: signal/signals.go (hunk 1 above)
Comment (divan) [reply signal/signals.go:49]: @adambabik no, I haven't. But as I understand, CCstring always make a copy and responsibility to free its memory is on the caller.

`javaJsonEvent` is a separate chunk of memory so it's ok that it is deleted there:
```c
    jstring javaJsonEvent = NULL;
    if (jsonEvent != NULL) {
        javaJsonEvent = (*env)->NewStringUTF(env, jsonEvent);
    }
```

**Review by gravityblast — APPROVED** [review]
Nice job!

**Review by dshulyak — APPROVED** [review]

**Comment by divan** [comment]
I'll test it on a real server built from the branch before merging.

**Comment by divan** [comment]
Update after running fix for ~20 hours. Memory increased from 17MB to 25MB (vs 17MB to 109MB for the same period before the fix).
I'm still not happy with CPU increase (which might be totally unrelated), so going to merge this..

screen shot 2018-07-26 at 12 29 47
